### DISCUSSION TITLE
Parsing Markdown in getStaticProps ?

### DISCUSSION THREAD

**Discussion by Delkathes** [vercel/next.js#16713] [Help]
Hi,

I'm exploring different ways to do markdown to jsx with ISG. I receive the markdown through `getStaticProps`.
(initial goal => rewrite `<a>` tags to `<NextLink />` for in-app smooth navigation)

But here I am with a question I can't solve.
Is there a benefit from parsing markdown strings in getStaticProps  ? (to mdast or hast)

I currently have two situations:

One with `remark-react`: Receive md string with `getStaticProps` => pass it as a prop => use `remark-react` to have jsx. In this situation, the md string isn't parsed in `getStaticProps` but directly rendered with `remark-react` in the app. This gives me a `console.timeEnd` of approx 20ms (up to 40). Haven't configured personal elements rewrites yet (`<a>` to `<Link />`, or `<img />` to `<CloudinaryImg  />`).

Second with own config: Receive md string with `getStaticProps` => parse it to mdast/hast => pass it as a prop =>use my own mdast/hast to jsx tool (untested etc ...). In this situation, the md string is parsed in `getStaticProps` to mdast/hast and rendered with my own tool in the app. This gives me a `console.timeEnd` of approx 1ms (0.9 to 1.3). Haven't configured personal elements rewrites yet (`<a>` to `<Link />`, or `<img />` to `<CloudinaryImg  />`).

I know 20ms isn't that big of a deal, but I'm trying to get the fastest thing possible, and need a better understanding of SSG and how things render.

Have a good day !

**Comment by rafaelalmeidatk** [comment]
You might want to try [`next-mdx-remote`](https://github.com/hashicorp/next-mdx-remote) (example [here](https://github.com/vercel/next.js/tree/canary/examples/with-mdx-remote)). It allows you to receive an external MDX string and load it on `gSP`, basically what you are doing with the second option
